### PR TITLE
add equalsIgnoreCase function to SafetyUtil

### PR DESCRIPTION
**Pull request by LenoreChen**
* add `equalsIgnoreCase()` 
* this function will be used in NANOPAY `Release-v3.17`

### CODE REVIEW

File: src/foam/util/SafetyUtil.java [hunk 1]
```diff
@@ -27,6 +27,10 @@ public static boolean equals(Object o1, Object o2) {
     return compare(o1, o2) == 0;
   }
 
+  public static boolean equalsIgnoreCase(String o1, String o2) {
```
Comment (jlhughes) [inline src/foam/util/SafetyUtil.java:30]: @kgrgreer do you want equalsIgnoreCase or equalsIC ? I know you like shorter names.

Comment (kgrgreer) [reply src/foam/util/SafetyUtil.java:30]: IC is more compatible with mLangs but IgnoreCase is compatible with the Java function, which this once mimics, so I think the way it is is correct

File: src/foam/util/SafetyUtil.java [hunk 2]
```diff
@@ -27,6 +27,10 @@ public static boolean equals(Object o1, Object o2) {
...
+    return o1.compareToIgnoreCase(o2) == 0;
```
Comment (kgrgreer) [inline src/foam/util/SafetyUtil.java:31]: This isn't safe because it isn't checking if o1 is null.

Comment (kgrgreer) [reply src/foam/util/SafetyUtil.java:31]: String's have an equalsIgnoreCase(), which is faster the compareTo().

**Comment by LenoreChen** [comment]
* address the null issue
* add tests for the `equalsIgnoreCase` function

**Pull request closed without merging**